mobile/src/components/courses/components/category-list/category-list.tsx

```diff
@@ -12,39 +19,48 @@ type Props = {
 };
 
 const CategoryList: FC<Props> = ({ items }) => {
-  const [activeCategoryIds, setActiveCategoryIds] = useState<number[]>([]);
+  const [activeCategoryId, setActiveCategoryId] = useState<number | null>(null);
   const [filteredCategories, setFilteredCategories] = useState<
     CategoryGetAllItemResponseDto[]
   >([]);
+  const dispatch = useAppDispatch();
```

Comment:
this logic should be outside CategoryList, you should pass handleSelect via props